pyiceberg/expressions/literals.py

```diff
@@ -145,6 +147,8 @@ def literal(value: L) -> Literal[L]:
         return BinaryLiteral(value)
     elif isinstance(value, Decimal):
         return DecimalLiteral(value)
+    elif isinstance(value, datetime):
+        return TimestampLiteral(datetime_to_micros(value))
```

Comment:
The `tzinfo` is handled here:

https://github.com/apache/iceberg-python/blob/3981e620f4560e3d13bff3d062bd3a89e39971d0/pyiceberg/utils/datetime.py#L75-L81